src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.EnumeratedValueSetFactory.cs

```diff
@@ -38,7 +39,14 @@ public IValueSet<T> Related(BinaryOperatorKind relation, T value)
 
             IValueSet<T> IValueSetFactory<T>.Random(int expectedSize, Random random)
             {
-                throw new NotImplementedException($"IValueSetFactory<{typeof(T).ToString()}>.Random");
+                TTC tc = default;
+                T[] values = tc.RandomValues(expectedSize, random, expectedSize * 2);
```

Comment:
> * 2 [](start = 79, length = 4)

Why `* 2`?

Reply:
So when you get 10 values, they are a random subset selected from a pool of 20 candidates (e.g. 10 values selected from [1..20])

---
In reply to: [383590170](https://github.com/dotnet/roslyn/pull/41912#discussion_r383590170) [](ancestors = 383590170)